conjurapi/host_factory.go

```diff
@@ -99,8 +104,8 @@ func (c *Client) CreateHost(data url.Values, token string) (HostFactoryHostRespo
 		return jsonResponse, err
 	}
 	err = json.Unmarshal(respData, &jsonResponse)
-	if err != nil {
-		return jsonResponse, err
+	if err == nil {
+		err = response.EmptyResponse(resp)
```

Comment:
If we get to this point in the function, then the prior call to `response.DataResponse(resp)` on line 102 has succeeded without error - calling `response.EmptyResponse(resp)` at this point is guaranteed to return `nil`.

Looking more at the `response` package, maybe we could use the `JSONResponse()` function instead of a combination of `response.DataResponse()` and `json.Unmarshal()`.